### PR TITLE
rc generation , vehicle module implementation

### LINKED ISSUE #3: RTO module  implementation

### PR DESCRIPTION
**Pull request by singlefleck**

### CODE REVIEW

File: lib/registration/customer.ex [hunk 1]
```diff
@@ -1,4 +1,7 @@
 defmodule Registration.Customer do
+  alias Registration.Helper
+  alias Static.StateCodes
+
   @spec is_valid(%{:aadhar => any, :name => binary, optional(any) => any}) :: nil | true
```
Comment (blackode) [inline lib/registration/customer.ex]: @spec return type is wrong here... 
change that to boolean

Comment (blackode) [inline lib/registration/customer.ex]:
```suggestion
  @spec is_valid(map()) :: boolean()
```

File: lib/registration/vehice_registration.ex [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    customer_info = Customer.get_customer_data()
+    vehicle_info = StateCodes.select_state()
+    vehicle_number = vehicle_number_generation(vehicle_info[:state])
+
+    name = customer_info[:name]
+    aadhar = customer_info[:aadhar]
+    state = vehicle_info[:state]
+    CliSpinners.spin()
+
+    rc_details = %{name: name, aadhar: aadhar, vehicle_number: vehicle_number, state: state}
```
Comment (blackode) [inline lib/registration/vehice_registration.ex:54]:
```suggestion
    rc_details = %{name: name, aadhar: aadhar, vehicle_number: vehicle_number, state: state}
    rc = rc_template(rc_details)
    IO.puts(rc)
```

File: lib/static/state_codes.ex [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+      Sikkim: "SK",
+      TamilNadu: "TN",
+      Tripura: "TR",
+      Uttar_Pradesh: "UP",
+      West_Bengal: "WB"
+    }
+
+    atom = String.to_atom(state)
+    Map.get(codes, atom)
+  end
```
Comment (blackode) [inline lib/static/state_codes.ex]:
```suggestion
```

File: lib/static/state_codes.ex [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+  @spec index_range(integer) :: tuple()
+  def index_range(n) do
+    if n < 6 do
+      low = n * 5 - 4
+      high = n * 5
+      {low, high}
+    else
+      "please start from first"
```
Comment (blackode) [inline lib/static/state_codes.ex]:
```suggestion
      {1, 5}
```

File: lib/static/state_codes.ex [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    end
+  end
+
+  @spec states_promt({integer, integer}) :: String.t()
+  @doc """
+  prepares the displable state prompt
+  """
+  def states_promt({low, high} = index_range) do
+    indexes = Enum.to_list(low..high)
+    code = get_state_code
```
Comment (blackode) [inline lib/static/state_codes.ex]:
```suggestion
    code = get_state_code()
```

File: mix.exs [hunk 6]
```diff
@@ -24,6 +24,9 @@ defmodule Registration.MixProject do
     [
       # {:dep_from_hexpm, "~> 0.3.0"},
       # {:dep_from_git, git: "https://github.com/elixir-lang/my_dep.git", tag: "0.1.0"}
+      {:ecto_sql, "~> 3.0"},
```
Comment (blackode) [inline mix.exs:27]:
```suggestion
       {:ecto, "~> 3.8"}
```

**Review by blackode — CHANGES_REQUESTED** [review]

File: mix.exs (hunk 6 above)
Comment (singlefleck) [reply mix.exs:27]: image

**Review by blackode — APPROVED** [review]

**Pull request merged**